### DISCUSSION TITLE
Ethereum Provider doesn't create a session after connecting

### DISCUSSION THREAD

**Discussion by tiennt0212** [WalletConnect/walletconnect-monorepo#2794] [Developer Support] [answered]
## Description
I have left my problem in another case since I thought it was related to my problem. However, I thought it's better if I have a separate ticket to say about that.

I'm using Ethereum Provider when migrating to WalletConnect 2.0. And this is my projected `b77a1626a05740d6a93b1707ac7b0761`

I'm used to connecting to WalletConnect successfully before. And I was able to sign a message, sign a transaction.

However, I cannot receive a session after connecting to WalletConnect for now.

## Actual result

After connecting to WalletConnect, there's no session was created.

Here are my screencasts (I removed it due to a personal issue)

## Expected result
After connecting to WalletConnect, a session was created for signing

**Comment by glitch-txs** [comment]
Does this issue happens with other wallets? Trust Wallet, Rainbow?

**Reply by tiennt0212** [reply, marked as answer]
It seems a problem with my chainId, I put it in a number like 80001 and it causes that error. When I changed into hexa "0x13881", it works.

My problem looks like to be resolved

**Reply by tiennt0212** [reply]
BTW, the new Wallet Connect infrastructure probably supports multiple chains. In my application, I would like to keep a single chain by using this initialization:

```
const provider = await EthereumProvider.init({
    projectId: PROJECT_ID,
    chains: [chainId],
    showQrModal: true,
    methods: [
      'eth_sendTransaction',
      'personal_sign',
      'eth_requestAccounts',
      'eth_signTypedData_v4',
    ],
    events: ['chainChanged', 'accountsChanged', 'connect', 'disconnect'],
  })
  ```
  
  
 However, when I try to change the chain, it likely appends a new chainId into the `chains` array. I don't know how to keep only one element in this array, or replace the chains[0] value.
  
@glitch-txs  Can you help me a solution? :sweat_smile:

**Reply by tiennt0212** [reply]
I'm able to listen to the 'chainChanged' event and receive the chainId. But I don't know how to update it to the connection :sweat_smile:

**Reply by glitch-txs** [reply]
You can't prevent the user from switching chain but you can set a default chain for a specific transaction by calling the `wallet_switchEthereumChain` method. It will set the chain automatically, you should see this reflected in the tx

**Reply by tiennt0212** [reply]
Oh, thanks for your suggestion, I'll try your way. I hope it will work for my application.

**Reply by finessevanes** [reply]
@tiennt0212 can you confirm you're on the latest ethereum-provider version 2.8.4?

**Reply by tiennt0212** [reply]
> can you confirm you're on the latest ethereum-provider version 2.8.4?

I confirmed I'm on the latest version 2.8.4 of @walletconnect/ethereum-provider

**Reply by finessevanes** [reply]
Awesome - are you able to update the session with the new chain?

**Reply by tiennt0212** [reply]
Hmm, do you mean the request "wallet_switchEthereumChain"? It looks to change the chain on WalletConnect only, but on my Metamask mobile application, it still doesn't have any requests. I will capture you a video.

In my expectation, a request will be displayed on the Metamask, when the user agrees to switch, the chain will be switch on WalletConnect also.

**Reply by tiennt0212** [reply]
Hmm, the video does not work as my expectation for a simulator... sorry.

In summary, I requested a `wallet_switchEthereumChain`. However, it doesn't display a request popup in Metamask, it just changes silently in Wallet Connect. It leads to the WalletConnect connecting to a chain, but my Metamask mobile application is connecting to another one.

**Reply by glitch-txs** [reply]
You won't receive a request since you're already connected on the wallet side. Keep in mind WalletConnect v2 is multi-chain

**Reply by tiennt0212** [reply]
Oh, thanks for your information, I'll test your scenario and let you know soon!

**Reply by glitch-txs** [reply]
You can check is working by triggering a transaction, you should see in the wallet the gas fee should be on eth in the mainnet case

**Reply by tiennt0212** [reply]
Pardon me, you mentioned _"you're already connected on the wallet side. Keep in mind WalletConnect v2 is multi-chain"_

I understand your sentence as:
- I expect to work on the chain A
- However, I connected to chain B when creating a session to WalletConnect.
- I'm able to connect to chain A by sending a request `wallet_switchEthereumChain`. Then, on Metamask mobile application will **display a request** for switching.

What I expect is a request displaying on Metamask mobile application :sweat_smile: Like the image below
<details>
<summary>IMAGE - Metamask displays change chain request</summary>

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/72179479/9bfd76bd-a949-4626-be65-0ad2b576e86c)

</details>

**Reply by tiennt0212** [reply]
So, I tested the above flow, and it doesn't work as I expected. It still changes the network silently without any displaying request

**Reply by glitch-txs** [reply]
You **won't** get a request, because there's nothing to request since you're already connected to that chain in the wallet side. You can be connected to multiple chains at once so "switching" is not necessary in the Wallet side, you only need to set a default chain on the dapp side to target a network when doing a transaction

**Reply by tiennt0212** [reply]
What a clear explanation! Thank you, I understood its idea
